src/screens/NanoContract/NanoContractRegisterScreen.js

```diff
@@ -0,0 +1,323 @@
+
+/**
+ * Validates the formModel, returning the invalidModel.
+ * If there is no error in the formModel, the invalidModel is returned empty.
+ */
+function validate(formModel) {
+  const invalidModel = {};
+
+  if (!formModel.ncId) {
+    invalidModel.ncId = t`Nano Contract ID is required.`;
+  }
+
+  return invalidModel;
+}
+
+export function NanoContractRegisterScreen({ navigation }) {
+  const dispatch = useDispatch();
+  const { address, error } = useSelector((state) => state.firstAddress);
+  const registerState = useSelector((state) => ({
+    registerStatus: state.nanoContract.registerStatus,
+    registerFailureMessage: state.nanoContract.registerFailureMessage,
+  }));
+
+  const [isClean, setClean] = useState(true);
+  const [formModel, setFormModel] = useState({
+    ncId: null,
+  });
+  const [invalidModel, setInvalidModel] = useState({
+    ncId: null,
+  });
+
+  /* @param {
+   *   'nodeUrl'|'explorerUrl'|'explorerServiceUrl'|'walletServiceUrl'|'walletServiceWsUrl'
+   * } name */
+  const handleInputChange = useCallback(
+    (name) => (value) => {
+      if (isClean) {
+        setClean(false);
+      }
+
+      // update invalid model
+      const invalidModelCopy = { ...invalidModel };
+      delete invalidModelCopy[name];
+      setInvalidModel(invalidModelCopy);
+
+      // update form model
+      const form = {
+        ...formModel,
+        [name]: value,
+      };
+      setFormModel(form);
+
+      // validate form model and update invalid model
+      setInvalidModel(validate(form));
+    },
+    [isClean, invalidModel, formModel]
+  );
+
+  const handleSubmit = useCallback(
+    () => {
+      const newInvalidModel = validate(formModel);
+      if (hasError(newInvalidModel)) {
+        setInvalidModel(newInvalidModel);
+        return;
+      }
+
+      const { ncId } = formModel;
+      dispatch(nanoContractRegisterRequest({ address, ncId }));
+    },
+    [formModel]
+  );
+
+  const handleFeedbackModalDismiss = () => {
+    dispatch(nanoContractRegisterReady());
+  };
+
+  const navigatesToNanoContractTransactions = () => {
+    dispatch(nanoContractRegisterReady());
+    navigation.replace('NanoContractDetailsScreen', { ncId: formModel.ncId });
+  };
+
+  useEffect(() => {
+    dispatch(firstAddressRequest());
+  }, []);
+
+  const hasFirstAddressFailed = () => error;
+  const isFirstAddressLoading = () => !error && !address;
+  const hasFirstAddressLoaded = () => !error && address;
+
+  return (
+    <Wrapper>
+      <NavigationHeader navigation={navigation} />
+
+      {hasSucceeded(registerState.registerStatus)
+        && (
+          <FeedbackModal
+            icon={(<Image source={checkIcon} style={styles.feedbackModalIcon} resizeMode='contain' />)}
+            text={feedbackSucceedText}
+            onDismiss={handleFeedbackModalDismiss}
+            action={(<NewHathorButton discrete title={t`See contract`} onPress={navigatesToNanoContractTransactions} />)}
+          />
```

Comment:
Can the user click outside the modal and just close it without the `action` to navigate to the other screen?